beacon-chain/blockchain/receive_attestation.go

```diff
@@ -202,10 +204,14 @@ func (s *Service) notifyEngineIfChangedHead(ctx context.Context, newHeadRoot [32
 	_, err = s.notifyForkchoiceUpdate(s.ctx, arg)
 	if err != nil {
 		log.WithError(err).Error("could not notify forkchoice update")
+		if IsInvalidBlock(err) {
```

Comment:
To not alter with original behavior too much, only return error if the block has invalid payload for blacklist

Reply:
This will log twice the same error, here and then in the outer caller when `err` surfaces. Perhaps consider putting this block inside an if-else block.